### DISCUSSION TITLE
Could not reach target 'foo' since it was skipped due to a criteria

### DISCUSSION THREAD

**Discussion by Marusyk** [cake-build/cake#3625] [Q&A] [answered]
I have something like
```
Task("UploadTestReport")
    .IsDependentOn("Test")
    .WithCriteria((context) => FileExists(testCoverageOutputFilePath))
    .WithCriteria(!string.IsNullOrWhiteSpace(coverallsRepoToken))
    .WithCriteria((context) => !BuildSystem.IsLocalBuild)
    .Does(() => { }
```
then my build failed with the message

> 
> Error: One or more errors occurred. (Could not reach target 'UploadTestReport' since it was skipped due to a criteria.)
> Could not reach target 'UploadTestReport' since it was skipped due to a criteria.
> Error: The process 'tools\dotnet-cake' failed with exit code 1

Is there a way how to know what exactly criteria don'twork?

**Comment by pascalberger** [comment]
You can define a message for each criteria, like in this example:

```
Task("A")
    .WithCriteria(() => DateTime.Now.Second % 2 == 0, "Need even seconds.")
    .Does(() => {

    });
```

See https://cakebuild.net/docs/writing-builds/tasks/criteria#print-message-from-criteria

**Reply by Marusyk** [reply]
thank you

**Reply by Marusyk** [reply]
it would be great to have the message not only when Verbosity is set to Verbose or Diagnostic. What do you think?

**Reply by Marusyk** [reply]
`dotnet cake --target UploadTestReport --verbosity=diagnostic`
Even thought, I don't see my message
![image](https://user-images.githubusercontent.com/9221694/139037549-9f751627-a980-4eac-845f-b31f8f735d48.png)

**Comment by devlead** [comment]
try
```
dotnet cake --target=UploadTestReport --verbosity=diagnostic
```

**Reply by Marusyk** [reply]
the same result:
dotnet cake --target=UploadTestReport --verbosity=diagnostic
![image](https://user-images.githubusercontent.com/9221694/139105318-cea0ae8b-7556-40aa-a704-6d2b8b3d3bb9.png)

**Reply by Marusyk** [reply]
`dotnet cake --target=UploadTestReport --verbosity=Verbose` also doesn't work.
Actually, it's work but my Error message is missing
![image](https://user-images.githubusercontent.com/9221694/139105837-3c24896e-cfdb-4a32-909b-b0c7cc331732.png)

**Comment by augustoproiete** [comment, marked as answer]
- Tracked via https://github.com/cake-build/cake/issues/3647
- Proposed fix via PR https://github.com/cake-build/cake/pull/3648
